Source/MediaInfo/Tag/File_Exif.cpp

```diff
@@ -1603,11 +2045,24 @@ void File_Exif::MulticodeString(ZtringList& Info)
 void File_Exif::Thumbnail()
 {
     Stream_Prepare(Stream_Image);
-    Fill(Stream_Image, 1, Image_Type, "Thumbnail");
-    Fill(Stream_Image, 1, Image_MuxingMode, "Exif");
 
-    File__Analyze* Parser = nullptr;
-    const auto Infos_Thumbnail_It = Infos.find(Kind_IFD1);
+    std::unique_ptr<File__Analyze> Parser;
+    kind_of_ifd IFD;
+
+    if (currentIFD == Kind_ParsingNikonPreview) {
+        Fill(Stream_Image, StreamPos_Last, Image_Type, "Preview Image");
```

Comment:
I see no practical difference between "Preview Image" and "Thumbnail", I would put "Thumbnail" for all.